docker/src/main/java/brooklyn/entity/container/docker/DockerHostImpl.java

```diff
@@ -481,6 +506,25 @@ public void deleteLocation() {
     protected void preStart() {
         getDriver().configureSecurityGroups();
 
+        // Save the VLAN id for this machine
+        MachineProvisioningLocation location = getInfrastructure().getDynamicLocation().getProvisioner();
+        if (isJcloudsLocation(location, SoftLayerConstants.SOFTLAYER_PROVIDER_NAME)) {
```

Comment:
a bit naughty to do this here!  can it go into `brooklyn-jclouds` project as a `SoftLayerSameVlanLocationCustomizer` ?

i think it would:

0. if vlan properties are set explicitly on the location, return
1. set `scope` = value of a location config `"SoftLayerSameVlanLocationCustomizer.scopeUid"` defined on the location 
2. in a `synchronized (SoftLayerSameVlanLocationCustomizer.class)` block (to prevent multiple threads creating the semaphore) get the location config `"SoftLayerSameVlanLocationCustomizer.scopeSemaphores."+scope`, or if not present set that to be a semaphore with one lock available
3. acquire the lock on the semaphore, blocking (calling `Tasks.setBlockingDetails("waiting on VLAN info")` before and `resetBlockingDetails` after)
4. look do we have any VM's in any machine provisioning location with the same scope, if so take the VLAN ID's from that location and set them on the provisioning request
5. return, without creating VM, then ensure we get a callback on creation *or on failure* and release the lock at that point

this should do the right thing, ie ensuring only one thread can create a VM if we don't yet have vlan details, and getting existing vlan details otherwise

/cc @aledsage the problem this solves is that SL may put your VM's into 

also update the `more-on-clouds.md` page in brooklyn docs to refer to this

ps @grkvlt this is an awesome idea i've nicked from you :)